receiver/statsdreceiver/transport/packet_server.go

```diff
@@ -0,0 +1,131 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package transport // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/transport"
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/multierr"
+)
+
+type packetServer struct {
+	packetConn net.PacketConn
+	transport  Transport
+	address    string
+}
+
+// Ensure that Server is implemented on Packet transport based servers.
+var _ (Server) = (*packetServer)(nil)
+
+// NewPacketServer creates a transport.Server using transports based on packets.
+func NewPacketServer(transport Transport, address string) (Server, error) {
```

Comment:
is address still the right term?

Reply:
Good question. I had issues finding new wording that covers an IP address and a socket path... socket address sounds good to me but I am open to suggestions.